nixos/modules/services/web-apps/peertube.nix

```diff
@@ -0,0 +1,357 @@
+      previews: '/var/lib/peertube/storage/previews/'
+      thumbnails: '/var/lib/peertube/storage/thumbnails/'
+      torrents: '/var/lib/peertube/storage/torrents/'
+      captions: '/var/lib/peertube/storage/captions/'
+      cache: '/var/lib/peertube/storage/cache/'
+      plugins: '/var/lib/peertube/storage/plugins/'
+      client_overrides: '/var/lib/peertube/storage/client-overrides/'
+
+    ${cfg.extraConfig}
+  '';
+
+  cfgService = {
+    # Access write directories
+    UMask = "0027";
+    # Capabilities
+    CapabilityBoundingSet = "";
+    # Security
+    NoNewPrivileges = true;
+    # Sandboxing
+    ProtectSystem = "strict";
+    ProtectHome = true;
+    PrivateTmp = true;
+    PrivateDevices = true;
+    PrivateUsers = true;
+    ProtectClock = true;
+    ProtectHostname = true;
+    ProtectKernelLogs = true;
+    ProtectKernelModules = true;
+    ProtectKernelTunables = true;
+    ProtectControlGroups = true;
+    RestrictNamespaces = true;
+    LockPersonality = true;
+    RestrictRealtime = true;
+    RestrictSUIDSGID = true;
+    PrivateMounts = true;
+    # System Call Filtering
+    SystemCallArchitectures = "native";
+  };
+
+in {
+  options.services.peertube = {
+    enable = lib.mkEnableOption "Enable Peertube’s service";
+
+    user = lib.mkOption {
+      type = lib.types.str;
+      default = "peertube";
+      description = "User account under which Peertube runs";
+    };
+
+    group = lib.mkOption {
+      type = lib.types.str;
+      default = "peertube";
+      description = "Group under which Peertube runs";
+    };
+
+    hostname = lib.mkOption {
+      type = lib.types.str;
+      default = "example.com";
+      description = "Server name of reverse proxy";
+    };
+
+    listenHttp = lib.mkOption {
+      type = lib.types.int;
+      default = 9000;
+      description = "listen port for HTTP server";
+    };
+
+    listenWeb = lib.mkOption {
+      type = lib.types.int;
+      default = 443;
+      description = "listen port for WEB server";
+    };
+
+    enableWebHttps = lib.mkOption {
+      type = lib.types.bool;
+      default = true;
+      description = "Enable or disable HTTPS protocol";
+    };
+
+    extraConfig = lib.mkOption {
+      type = lib.types.lines;
+      default = "";
+      example = ''
+        listen:
+          hostname: '0.0.0.0'
+        trust_proxy:
+          - '192.168.10.21'
+        log:
+          level: 'debug'
+      '';
+      description = "Extra config options for peertube";
+    };
+
+    redis = {
+      createLocally = lib.mkOption {
+        type = lib.types.bool;
+        default = true;
+        description = "Configure local Redis server for PeerTube.";
+      };
+
+      host = lib.mkOption {
+        type = lib.types.str;
+        default = "127.0.0.1";
+        description = "Redis host.";
+      };
+
+      port = lib.mkOption {
+        type = lib.types.port;
+        default = 6379;
+        description = "Redis port.";
```

Comment:
Why not socket?

Reply:
The socket is specified by the parameter `redis.enableUnixSocket`. Used by default.